protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java

```diff
@@ -142,20 +142,23 @@ public void wireExtensions(Class interfaceName, List extensions) throws WiringEx
     
     @Override
     public Response onLine(SessionT session, ByteBuffer line) {
-        
+        Request request;
         try {
             
-            Request request = parseRequest(session, line);
+            request = parseRequest(session, line);
             if (request == null) {
                 return null;
             }
-            return dispatchCommandHandlers(session, request);
         } catch (Exception e) {
             LOGGER.debug("Unable to parse request", e);
             return session.newFatalErrorResponse();
-        } 
-
-       
+        }
+        try {
+            return dispatchCommandHandlers(session, request);
+        } catch (Exception e) {
+            LOGGER.error("Error dispatching command for request {}", request, e);

```

Comment:
yes I tested my code :) the request is well printed. Do you want to mean more by your comment?

looking at the code, Request has too implementations. BaseRequest (which should be used by smtp) has a toString() method. But ImapRequest has not. I'm not very confident adding it, because we could be lead to disclose passwords in logs. Maybe even request should not be logged, or at least not the arguments?

Reply:
if you agree I will log only request.getCommand()

Reply:
I agree with it, or you can create a method like `asString()` to provide enough information of the each `Request` impl.

Reply:
ok, changed to request.getCommand, because I'm not very confident introducing such an API change in Request interface.